src/client/Microsoft.Identity.Client/Internal/Requests/SlcManagedIdentityAuthRequest.cs

```diff
@@ -0,0 +1,318 @@
+﻿// Copyright (c) Microsoft Corporation. All rights reserved.
+// Licensed under the MIT License.
+
+using System;
+using System.Collections.Generic;
+using System.Net.Http;
+using System.Threading;
+using System.Threading.Tasks;
+using Microsoft.Identity.Client.ApiConfig.Parameters;
+using Microsoft.Identity.Client.AppConfig;
+using Microsoft.Identity.Client.Cache.Items;
+using Microsoft.Identity.Client.Core;
+using Microsoft.Identity.Client.Http;
+using Microsoft.Identity.Client.OAuth2;
+using Microsoft.Identity.Client.PlatformsCommon.Interfaces;
+using Microsoft.Identity.Client.Utils;
+using Microsoft.Identity.Client.ManagedIdentity;
+using System.Linq;
+
+namespace Microsoft.Identity.Client.Internal.Requests
+{
+    internal class SlcManagedIdentityAuthRequest : ManagedIdentityAuthRequest
```

Comment:
Same comment as earlier - we probably should not expose the SLC name in the API surface. How about `ManagedIdentityAuthRequestion` (if the other one is legacy)?